src/types/ffi_url.rs

```diff
@@ -0,0 +1,42 @@
+use uniffi::deps::bytes::Buf;
+
+use crate::prelude::*;
+use crate::UniFfiTag;
+
+/// A wrapper around `Url` that allows us to safely deal with Urls generated on hosts.
+///
+/// Context: We have defined a custom type conversion between Rust's `Url` and the hosts analogues.
+/// However, a Url could be considered valid on host but not on Rust. For example, Swift allows to build a Url
+/// from string `"invalid input"`, while Rust doesn't.
+///
+/// Therefore, if a given Rust function expects a `Url` as param and is sent one from host side which is invalid,
+/// the code will panic. However, if we send the wrapper instead, we make sure the conversion is safely done on the
+/// host side, dealing with the failing conversion properly rather than panicking.
+#[derive(Debug, PartialEq, Eq, Hash, uniffi::Object, derive_more::Display)]
+#[uniffi::export(Debug, Display, Eq, Hash)]
+pub struct FfiUrl {
+    pub url: Url,
+}
+
+#[uniffi::export]
+impl FfiUrl {
```

Comment:
can probably make this `impl FromStr for FfiUrl` and do `Url::parse(s).map_err(..).map(SUT::new)` ? which results in a bit nicer tests both here and in `src/profile/v100/app_preferences/app_preferences_uniffi_fn.rs`

Reply:
Added it to clean up the `app_preferences_uniffi_fn.rs` test, but we still want to send a `Url` to this file test as we compare both `Url` match.

Therefore, by adding it to help clean up one test, I ended up needing to add an extra test on this file (to cover a function that is only used in tests). Wonder if that actually helps or adds more noise than before.

Reply:
Your call :) but true less is
More. I tend to impl too many traits 😑